sqlglot/optimizer/optimize_joins.py

```diff
@@ -59,11 +84,20 @@ def reorder_joins(expression):
     """
     for from_ in expression.find_all(exp.From):
         parent = from_.parent
-        joins = {join.alias_or_name: join for join in parent.args.get("joins", [])}
-        dag = {name: other_table_names(join) for name, join in joins.items()}
+        joins = parent.args.get("joins", [])
+
+        if not _is_reorderable(joins):
```

Comment:
I don't know the full context on why this rule was added (I'm assuming for normalization), but is it worth reordering subsets of the joins?

i.e.

```
SELECT
  * 
FROM x 
CROSS JOIN y 
JOIN z 
  ON x.a = z.a 
  AND y.a = z.a
LEFT JOIN q
  ON z.b = q.b
```

I think the first two joins can still be reordered?